sw/device/lib/testing/sram_ctrl_testutils.h

```diff
@@ -20,35 +20,43 @@
 typedef uint32_t (*sram_ctrl_testutils_exec_function_t)(volatile uint32_t,
```

Comment:
Hang on, why are the arguments to this function `volatile`? Moreover, note that `volatile` (like `const`) is meaningless at the top-level of a type in a function declaration (including function pointer types).

Moreover, `volatile` in a function parameter in this way is particularly meaningless at the definition site, for the same reason that `volatile` on an automatic storage duration variable is mostly meaningless without justification.

Reply:
Good point, this function is type is completely redundant now with the #9489 . I will remove it altogether.

Reply:
Done in the above mentioned PR.